File: java/src/main/java/com/svix/agentic_commerce_protocol/ErrorCode.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum ErrorCode implements ToQueryParam {
    MISSING("missing"),
    INVALID("invalid"),
    OUT_OF_STOCK("out_of_stock"),
    PAYMENT_DECLINED("payment_declined"),
    REQUIRES_SIGN_IN("requires_sign_in"),
    REQUIRES_3DS("requires_3ds");
    private final String value;

    ErrorCode(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/ErrorCode.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class ErrorCode : ToQueryParam {
    @SerialName("missing") MISSING,
    @SerialName("invalid") INVALID,
    @SerialName("out_of_stock") OUT_OF_STOCK,
    @SerialName("payment_declined") PAYMENT_DECLINED,
    @SerialName("requires_sign_in") REQUIRES_SIGN_IN,
    @SerialName("requires_3ds") REQUIRES_3DS;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
